### PR TITLE
Add logging total lock time

### PR DESCRIPTION
**Pull request by jborrey**
It is useful to know the total time that a lock was acquired for.

### CODE REVIEW

**Comment by jborrey** [comment]
Alternative to using hashes in the logger would be to have 2 modes, one for strings and the other for hashes. But adding this complexity does not seem worth it.

File: lib/master_lock.rb [hunk 1]
```diff
@@ -83,15 +85,21 @@ def synchronize(key, options = {})
...
         @registry.register(lock, extend_interval)
-      logger.debug("Acquired lock #{key}")
+      logger.debug(key: key, message: 'Acquired lock')
       begin
         yield
       ensure
         @registry.unregister(registration)
+
+        log_info = {
+          key: key,
```
Comment (jimpo) [inline lib/master_lock.rb]: "key" will get censored by the log pipeline. Call it "lock_id" instead.

File: lib/master_lock.rb [hunk 2]
```diff
@@ -56,6 +56,8 @@ class << self
     # @raise [LockNotAcquiredError] if the lock cannot be acquired before the
     #   timeout
     def synchronize(key, options = {})
+      start_time = Time.now # used for benchmarking
```
Comment (jimpo) [inline lib/master_lock.rb]: What time is this? Do you want the amount of time the lock was held for or how long it took to acquire it?

File: README.md [hunk 3]
```diff
@@ -38,6 +38,19 @@ MasterLock.start
 
 See [documentation](http://www.rubydoc.info/gems/master_lock) for advanced usage.
 
+## Example Config
+
+```ruby
+MasterLock.configure do |config|
+  config.redis = $redis
+  config.logger.level = Rails.env.test? ? :info : :debug
```
Comment (jimpo) [inline README.md]: Remove the Rails reference. Makes it seem like this library depends on Rails.

File: README.md [hunk 4]
```diff
@@ -38,6 +38,19 @@ MasterLock.start
...
+
+  config.logger.formatter = proc do |_serverity, _time, _progname, msg|
```
Comment (jimpo) [inline README.md]: Add explicit text above that the log formatter must accept hashes.

File: spec/spec_helper.rb [hunk 5]
```diff
@@ -68,7 +68,3 @@
     clean_redis
   end
 end
-
-MasterLock.configure do |config|
```
Comment (jimpo) [inline spec/spec_helper.rb]: Why remove this?

**Review by jimpo — CHANGES_REQUESTED** [review]

File: lib/master_lock.rb (hunk 2 above)
Comment (jborrey) [reply lib/master_lock.rb]: amount of time the lock was held for.

File: lib/master_lock.rb (hunk 1 above)
Comment (jborrey) [reply lib/master_lock.rb]: was working - but will do. i can see that happening elsewhere.

File: README.md (hunk 3 above)
Comment (jborrey) [reply README.md]: yeah fair call.

File: README.md (hunk 4 above)
Comment (jborrey) [reply README.md]: okay

File: spec/spec_helper.rb (hunk 5 above)
Comment (jborrey) [reply spec/spec_helper.rb]: oh my bad - was messing with tests too much. will put back

File: lib/master_lock.rb [hunk 6]
```diff
@@ -56,6 +56,8 @@ class << self
     # @raise [LockNotAcquiredError] if the lock cannot be acquired before the
     #   timeout
     def synchronize(key, options = {})
+      lock_acquired_start_time = Time.now # used for benchmarking
```
Comment (jimpo) [inline lib/master_lock.rb:59]: The lock has not been acquired yet. I don't think you are measuring the right thing.

**Comment by jimpo** [comment]
Closing due to inactivity.

**Pull request closed without merging**